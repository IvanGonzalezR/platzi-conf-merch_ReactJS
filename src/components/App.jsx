import React from "react";
import '../styles/components/App.scss';

const App = () => {
  return (
    <h1>App</h1>
  )
}

export { App };
