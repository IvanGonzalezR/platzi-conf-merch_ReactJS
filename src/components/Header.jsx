import React from 'react'
import '../styles/components/Header.css'
import { NavLink } from 'react-router-dom'
import { AppContext } from '../context/AppContext'

const Header = () => {

  const { state } = React.useContext(AppContext);
  const { cart } = state;

  return (
    <div className="Header">
      <NavLink
        to="/"
        end
      >
        <h1 className="Header-title">PlatziConf Merch</h1>
      </NavLink>
      <div className="Header-checkout">
        <NavLink
          // style={({ isActive }) => ({
          //   color: isActive ? 'black' : 'blue'
          // })}
          to="/checkout"
          end
        >
          <i className='fas fa-shopping-basket'></i>
        </NavLink>
        {cart.length > 0 ?
          <div className='Header-alert'>{cart.length}</div>
          : null
        }
      </div>
    </div>
  )
}

export { Header };
