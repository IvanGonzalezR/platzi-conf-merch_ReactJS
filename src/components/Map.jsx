import React from 'react'
import '../styles/components/Map.css'
import { MapContainer, TileLayer, Marker } from 'react-leaflet';

const MAPSAPIKEY = "AIzaSyDEAT52oSUDjufgGAIchIETg1NNqahK-U4";

const Map = ({ data }) => {
  console.log('LOCATION')
  console.log(data);
  return (

    <div className='leaflet-container'>
      <link
        rel="stylesheet"
        href="https://unpkg.com/leaflet@1.6.0/dist/leaflet.css"
        integrity="sha512-xwE/Az9zrjBIphAcBb3F6JVqxf46+CDLwfLMHloNu6KEQCAWi6HcDUbeOfBIptF7tcCzusKFjFw2yuvEpDL9wQ=="
        crossOrigin=""
      />

      <MapContainer center={[ data.latitude, data.longitude ]} zoom={16} scrollWheelZoom={true}>
        <TileLayer
          attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
        />
        <Marker position={[ data.latitude, data.longitude ]}>
          <i className="fa-solid fa-location-dot"></i>
        </Marker>
      </MapContainer>
    </div>
  );
}

export { Map };
