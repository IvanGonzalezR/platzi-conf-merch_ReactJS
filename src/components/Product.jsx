import React from 'react'

const Product = ({ product, handleAddToCart }) => {
  const [ label, setLabel ] = React.useState('Comprar');

  const changeLabel = () => {
    label === 'Comprar'
      ? setLabel('Vender')
      : setLabel('Comprar')
  }
  return (
    <div className='Products-item'>
      <img src={product.image} alt={product.title} />
      <div className='Products-item-info'>
        <h2>
          {product.title}
          <span>
            ${product.price}
          </span>
        </h2>
        <p>{product.description}</p>
      </div>
      <button type='button' onClick={() => {
        handleAddToCart(product);
        changeLabel();
      }}>{label}</button>
    </div>
  )
}

export { Product };
