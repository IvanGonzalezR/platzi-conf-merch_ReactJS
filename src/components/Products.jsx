import React from 'react'
import '../styles/components/Products.css'
import { Product } from './Product'
import { AppContext } from '../context/AppContext'

const Products = () => {

  const { state: { products }, addToCart } = React.useContext(AppContext);

  const handleAddToCart = (product) => {
    addToCart(product);
  }

  return (
    <div className='Products'>
      <div className='Products-items'>
        {products.map((product) => (
          <Product
            key={product.id}
            product={product}
            handleAddToCart={handleAddToCart}
          ></Product>
        ))}
      </div>
    </div>
  )
}

export { Products };
