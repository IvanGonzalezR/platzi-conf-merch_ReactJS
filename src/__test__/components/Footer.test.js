import React from "react";
import TestRenderer from 'react-test-renderer';
import { Footer } from "../../components/Footer";

describe("<Footer />", () => {
  const footer = TestRenderer.create(<Footer />);

  it("should render the footer", () => {
    expect(footer).toMatchSnapshot();
  });
  it('Render del titulo', () => {
    const tree = footer.toJSON();
    //Assert
    expect(tree.children[ 0 ].children[ 0 ]).toContain('PlatziConf');
    expect(tree.children[ 1 ].children[ 0 ]).toContain('os');
  });
});

describe('Footer Snapshot', () => {
  const footer = TestRenderer.create(<Footer />);
  const tree = footer.toJSON();
  test('Comprobar la UI del componente Footer', () => {
    expect(tree).toMatchSnapshot();
  });
});