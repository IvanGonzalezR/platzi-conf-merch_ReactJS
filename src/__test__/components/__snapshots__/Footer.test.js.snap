// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render the footer 1`] = `
<div
  className="Footer"
>
  <p
    className="Footer-title"
  >
     PlatziConf Merch 
  </p>
  <p
    className="Footer-copy"
  >
     Todos los derechos reservados 
  </p>
</div>
`;

exports[`Footer Snapshot Comprobar la UI del componente Footer 1`] = `
<div
  className="Footer"
>
  <p
    className="Footer-title"
  >
     PlatziConf Merch 
  </p>
  <p
    className="Footer-copy"
  >
     Todos los derechos reservados 
  </p>
</div>
`;
